pkg/virt-api/webhooks/validating-webhook/admitters/validate-k8s-utils.go

```diff
@@ -53,6 +55,30 @@ var nodeFieldSelectorValidators = map[string]func(string, bool) []string{
 	metav1.ObjectNameField: ValidateNodeName,
 }
 
+// validateSpecHostName is function that validate spec.hostName and checks if given affinities are valid
+// instead of bring in the whole kubernetes lib we simply copy it from kubernetes/pkg/apis/core/validation/validation.go
+func validateSpecHostName(fldPath *field.Path, hostName string) (causes []metav1.StatusCause) {
+	if hostName == "" {
+		return
+	}
+
+	errorList := field.ErrorList{}
+	for _, msg := range validation.IsDNS1123Label(hostName) {
+		errorList = append(errorList, field.Invalid(fldPath.Child("hostname"), hostName, msg))
+	}
+
+	//convert errorList to []metav1.StatusCause
+	for _, validationErr := range errorList {
+		causes = append(causes, metav1.StatusCause{
```

Comment:
can you please move the function `appendNewStatusCauseForHostNameNotConformingToDNSLabelRules` to this utils file and use the same format of error message?  we should not change the error message for existing stable APIs